chromium_src/components/payments/content/payment_request.cc

```diff
@@ -0,0 +1,111 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/payments/content/payment_request.h"
+
+#include "base/bind.h"
+#include "brave/browser/brave_rewards/rewards_service_factory.h"
+#include "brave/components/brave_rewards/common/constants.h"
+#include "brave/components/payments/content/buildflags/buildflags.h"
+#include "content/public/browser/browser_context.h"
+#include "third_party/blink/public/mojom/payments/payment_request.mojom.h"
+
+using payments::mojom::PaymentErrorReason;
+
+#if BUILDFLAG(ENABLE_PAY_WITH_BAT)
+namespace payments {
+
+void PaymentRequest::GetPublisherDetailsCallback(
+    mojo::PendingRemote<mojom::PaymentRequestClient> client,
+    std::vector<mojom::PaymentMethodDataPtr> method_data,
+    mojom::PaymentDetailsPtr details,
+    mojom::PaymentOptionsPtr options,
+    const ledger::type::Result result,
+    ledger::type::PublisherInfoPtr info) {
+  if (!info || info->status == ledger::type::PublisherStatus::NOT_VERIFIED) {
+    log_.Error(brave_rewards::errors::kInvalidPublisher);
+    TerminateConnection();
+    return;
+  }
+
+  Init_ChromiumImpl(std::move(client), std::move(method_data),
+                    std::move(details), std::move(options));
+}
+
+void PaymentRequest::TerminateConnectionWithMessage(PaymentErrorReason reason,
+                                                    std::string err) {
+  client_->OnError(reason, err);
+  TerminateConnection();
+}
+
+void PaymentRequest::Init(
+    mojo::PendingRemote<mojom::PaymentRequestClient> client,
+    std::vector<mojom::PaymentMethodDataPtr> method_data,
+    mojom::PaymentDetailsPtr details,
+    mojom::PaymentOptionsPtr options) {
+  bool result = false;
+  content::WebContents* contents = web_contents();
+  content::BrowserContext* context = contents->GetBrowserContext();
+
+  for (size_t i = 0; i < method_data.size(); i++) {
```

Comment:
Nit: Can this use a range-based for?

Reply:
discussed offline, `PaymentMethodDataPtr ` uses `DISALLOW_COPY_AND_ASSIGN` that forces us to use the conventional for loop.

Reply:
what if you try `(const auto& method : method_data)` ?

Reply:
Let me know if the code looks a bit better now: https://github.com/brave/brave-core/pull/7557/files#diff-12d68f7aee6cfd7305f59bcf92a37b1fa2353b1bd0a8b5224c7f4b63c68bfd2bR26